### PR TITLE
PP-711 add clientid and clientsecret credentials

### PR DESCRIPTION
**Pull request by igor-gini**

### CODE REVIEW

**Review by zladzeyka — CHANGES_REQUESTED** [review]
Thank you @igor-gini 👏🏻👏🏻
Could you please check the review notes?
May I ask you additionally to the review notes to look at possible refactoring of `SettingsViewController. configureTableView()`?

File: BankSDK/GiniBankSDKExample/GiniBankSDKExample/APIEnvironment.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+            return .custom(domain: "pay-api.pia.stage.gini.net", tokenSource: nil)
+        }
+    }
+
+    public var userApi: UserDomain {
+        switch self {
+        case .production:
+            return .default
+        case .stage:
+            return .custom(domain: "pay-api.pia.stage.gini.net")
```
[the remark this answers is not in the document]
Comment (igor-gini) [reply BankSDK/GiniBankSDKExample/GiniBankSDKExample/APIEnvironment.swift]: Fixed

File: BankSDK/GiniBankSDKExample/Tests/SettingsViewControllerTests.swift [hunk 2]
```diff
@@ -82,7 +82,8 @@ final class SettingsViewControllerTests: XCTestCase {
 	
 	override func setUp() {
 		super.setUp()
-		settingsViewController = SettingsViewController(giniConfiguration: configuration,
+        settingsViewController = SettingsViewController(apiEnvironment: .stage,
```
[the remark this answers is not in the document]
Comment (igor-gini) [reply BankSDK/GiniBankSDKExample/Tests/SettingsViewControllerTests.swift]: Switched to prod

File: BankSDK/GiniBankSDKExample/Tests/ScreenAPICoordinatorTests.swift [hunk 3]
```diff
@@ -20,7 +20,8 @@ final class ScreenAPICoordinatorTests: XCTestCase {
     var screenAPICoordinator: ScreenAPICoordinator?
     
     func testInitialization() {
-        screenAPICoordinator = ScreenAPICoordinator(configuration: GiniBankConfiguration(),
+        screenAPICoordinator = ScreenAPICoordinator(apiEnvironment: .stage,
```
[the remark this answers is not in the document]
Comment (igor-gini) [reply BankSDK/GiniBankSDKExample/Tests/ScreenAPICoordinatorTests.swift]: Fixed

File: BankSDK/GiniBankSDKExample/GiniBankSDKExample/SettingsViewController/SettingsViewController.swift [hunk 4]
```diff
@@ -543,18 +582,38 @@ extension SettingsViewController: SegmentedOptionTableViewCellDelegate {
...
 	}
 }
 
+// MARK: - CredentialsTableViewCellDelegate
+
+extension SettingsViewController: CredentialsTableViewCellDelegate {
+    func didTapSaveButton(clientId: String, clientSecret: String) {
+        dismissKeyboard()
+        delegate?.didTapSaveCredentialsButton(clientId: clientId, clientSecret: clientSecret)
+    }
```
[the remark this answers is not in the document]
Comment (igor-gini) [reply BankSDK/GiniBankSDKExample/GiniBankSDKExample/SettingsViewController/SettingsViewController.swift:622]: Exactly, the client id and secret are applied on the save button press

File: BankSDK/GiniBankSDKExample/GiniBankSDKExample/SettingsViewController/SettingsViewController.swift [hunk 5]
```diff
@@ -543,18 +582,38 @@ extension SettingsViewController: SegmentedOptionTableViewCellDelegate {
 		guard case .segmentedOption(var data) = option else { return }
         data.selectedIndex = cell.selectedSegmentIndex
         
-        switch data.selectedIndex {
+        if data is FileImportSegmentedOptionModel {
```
[the remark this answers is not in the document]
Comment (igor-gini) [reply BankSDK/GiniBankSDKExample/GiniBankSDKExample/SettingsViewController/SettingsViewController.swift:585]: Sure, separated into two methods

File: BankSDK/GiniBankSDKExample/GiniBankSDKExample/SettingsViewController/SettingsViewController+SegmentedOptionModel.swift [hunk 6]
```diff
@@ -5,7 +5,15 @@
 //  Created by Valentina Iancu on 16.06.23.
```
[the remark this answers is not in the document]
Comment (igor-gini) [reply BankSDK/GiniBankSDKExample/GiniBankSDKExample/SettingsViewController/SettingsViewController+SegmentedOptionModel.swift]: Fixed

**Review by zladzeyka — APPROVED** [review]
Thanks @igor-gini 
I've fixed the link to the user stage API 🥹

**Pull request merged**